File: statistics/population/mean/java/mean.java
// Java class for calclutating the mean of the population

/*
Input should be in the following format :
    number of elements in populatin : n
    elements : [space separeted n elements]
*/

import java.util.Scanner;

public class Mean{

    public static double calculateMean(int[] population, int n){

        int sum  = 0;
        for(int i : population){
            sum += i;
        }

        return (double) sum / n;
    }

    public static void main(String [] args){

        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        int [] population = new int[n];
        for(int i = 0;i<n;i++){
            population[i] = scanner.nextInt();
        }

        double mean = calculateMean(population,n);
        System.out.println(mean);
    }
}

File: statistics/population_mean/golang/mean.go
package main

import (
	"fmt"
)

func mean(numbers []int) float64 {
	sum := 0.0
	for _, number := range numbers {
		sum = sum + float64(number)
	}
	return sum / float64(len(numbers))
}

func main() {
	numbers := []int{2, 3, 5, 7, 11, 13}
	fmt.Println(mean(numbers))
}
